Repository/AbstractPhpcrRepository.php

```diff
@@ -110,4 +115,39 @@ protected function unresolvePath($path)
      * @return ArrayResourceCollection
      */
     abstract protected function buildCollection(array $nodes);
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getVersions($path)
+    {
+        try {
+            return new VersionList($path, [$this->get($path)]);
+        } catch (ResourceNotFoundException $e) {
+            throw NoVersionFoundException::forPath($path, $e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function remove($query, $language = 'glob')
+    {
+        Assert::notEq('', trim($query, '/'), 'The root directory cannot be deleted.');
+        Assert::startsWith($query, '/', 'The target path %s is not absolute.');
+
+        $resolvedPath = $this->resolvePath($query);
+
+        $this->removeResource($resolvedPath, $deleted);
```

Comment:
Can we change this to:

```php
return $this->removeResource($resolvedPath);
```

and have `removeResource` simply return the number of items deleted?